yt_idv/scene_components/base_component.py

```diff
@@ -83,17 +83,23 @@ def render_gui(self, imgui, renderer, scene):
             if len(self.iso_layers) < 32:
                 changed = True
                 self.iso_layers.append(0.0)
-        if imgui.button("Remove Layer"):
-            if len(self.iso_layers) > 0:
-                changed = True
-                self.iso_layers.pop()
-        for i in range(len(self.iso_layers)):
-            changed, self.iso_layers[i] = imgui.input_float(
+        imgui.columns(2, "iso_layers_cols", False)
+        i = 0
+        while i < len(self.iso_layers):
```

Comment:
I'm trying to figure out how to manage this without a `while` loop, but I can't quite come up with something.  So I suppose, this is OK!  :)

Reply:
I also think this is good for now! But I also agree that there could a way to simplify this... maybe separating the initial table construction from row removal when a button is pushed? Would result in more code but might be conceptually simpler. But ya, I say leave this as is for now.